pkgs/by-name/al/alist/package.nix

```diff
@@ -0,0 +1,83 @@
+{
+  lib,
+  alist,
+  buildGoModule,
+  fetchFromGitHub,
+  fetchurl,
+  fuse,
+  go,
+  stdenv,
+  installShellFiles,
+  testers,
+}:
+buildGoModule rec {
+  pname = "alist";
+  version = "3.38.0";
+
+  src = fetchFromGitHub {
+    owner = "alist-org";
+    repo = "alist";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-1nqveTEYrN9kYAOMTbLPw0MFNNNbFOqej+d+aTLAZ1o=";
+  };
+  web = fetchurl {
+    url = "https://github.com/alist-org/alist-web/releases/download/${version}/dist.tar.gz";
+    hash = "sha256-jHbWhjvHfgtdocuuELbOwrMzL8tOQfBVdH9MxasEwGo=";
+  };
+
+  CGO_ENABLED = 1;
+
+  vendorHash = "sha256-qSbvb2/y5rdS/OCutNEcRDUQBCAgNudux8XDnY9TRSo=";
+
+  buildInputs = [ fuse ];
+
+  tags = [ "jsoniter" ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X \"github.com/alist-org/alist/v3/internal/conf.GitAuthor=Xhofe <i@nn.ci>\""
+    # time should not be used as the input for nix, use "Nix" as a placeholder instead.
+    "-X github.com/alist-org/alist/v3/internal/conf.BuiltAt=Nix"
+    "-X github.com/alist-org/alist/v3/internal/conf.GoVersion=${go.version}"
+    "-X github.com/alist-org/alist/v3/internal/conf.GitCommit=${version}"
+    "-X github.com/alist-org/alist/v3/internal/conf.Version=${version}"
+    "-X github.com/alist-org/alist/v3/internal/conf.WebVersion=${version}"
+  ];
+
+  preConfigure = ''
+    # build sandbox do not provide network
+    rm pkg/aria2/rpc/client_test.go pkg/aria2/rpc/call_test.go
+
+    # use matched web files
+    rm -rf public/dist
+    cp ${web} dist.tar.gz
+    tar -xzf dist.tar.gz
+    mv -f dist public
+  '';
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    installShellCompletion --cmd alist \
+      --bash <($out/bin/alist completion bash) \
+      --fish <($out/bin/alist completion fish) \
+      --zsh <($out/bin/alist completion zsh)
+  '';
+
+  passthru.tests = {
+    version = testers.testVersion {
+      package = alist;
+      command = "${lib.getExe alist} version";
+    };
+  };
+
+  meta = {
+    description = "File list/WebDAV program that supports multiple storages";
+    homepage = "https://github.com/alist-org/alist";
+    changelog = "https://github.com/alist-org/alist/releases/tag/v${version}";
+    license = lib.licenses.agpl3Plus;
```

Comment:
I do not find any license notices specifying `agpl3Plus` or `agpl3Only`.  It is a common problem for projects hosted on Github.  Please contract upstream which license is used.

https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206#how-to-determine-license-3

https://www.gnu.org/licenses/identify-licenses-clearly.html

---

In addition, the license of `web` is unclear.  Please contract upstream.

Reply:
https://github.com/AlistGo/alist/issues/7346

Reply:
It seems the license of `web` is MIT: https://github.com/AlistGo/alist-web/blob/590eb0692cd8eaf2247798fc3ad42b818f6b7214/package.json#L35.

About `agpl3Plue` and `agpl3Only`, I think it is fine to merge this PR as is and update it if needed in the future.